### ISSUE TITLE
[sandboxie plus] Restore immediate recovery as in Classic layout with a list of folders used recently

### ISSUE THREAD

**Issue by diieg707** [sandboxie-plus/Sandboxie#988]
The immediate recovery prompt is not very friendly when it comes to recover all files in a folder.
so my request is that you could make it a little more like the "file recovery window", with a folder and subfolder tree format, following my other request #987 

![Capture](https://user-images.githubusercontent.com/54504707/125178684-8f0d1580-e1bd-11eb-9875-fda26aa03689.PNG)

**Comment by DavidXanatos** [comment]
you can open the file recovery window instead of using the "immediate recovery prompt" which here is just a feature of the generic notification window

**Comment by diieg707** [comment]
> 
> 
> you can open the file recovery window instead of using the "immediate recovery prompt" which here is just a feature of the generic notification window

but the folder/subfolder tree format is better to immediately recover lots of files at once, like when we are extracting something

**Comment by diieg707** [comment]
**man honestly you need to change this immediate recovery window, it's bad when it comes to recover all files in a folder at once**,
if you make it look like recovery window **(+ show subfolder tree)** it would be great.

![Image 2](https://user-images.githubusercontent.com/54504707/126838921-39c0f02e-bbf7-4879-ab2b-55206eeb220d.jpg)

**Comment by DavidXanatos** [comment]
How is this: 
![grafik](https://user-images.githubusercontent.com/3890945/129324542-bd4c65ba-962c-489b-8faa-a104dda21514.png)

better than the current scheme?
are you missing the ability to recover all the files at once?

so you want the behavioure of clicking the recovery -> open recovery dialog option?

So for your usecase you would want a switch to make the full recovery window pop up instead of showing only the notification in the notifications window?

**Comment by DavidXanatos** [comment]
In the next build you will have the option
![image](https://user-images.githubusercontent.com/3890945/129348051-549ec17a-4151-4dbd-b9d8-a9908a30581a.png)

with it sbie+ wil show the full recovery window wheneer a file is ready to be reocvered instead of adding it to the notification window

**Comment by diieg707** [comment]
> 
> 
> How is this:
> ![grafik](https://user-images.githubusercontent.com/3890945/129324542-bd4c65ba-962c-489b-8faa-a104dda21514.png)
> 
> better than the current scheme?
> are you missing the ability to recover all the files at once?
> 
> so you want the behavioure of clicking the recovery -> open recovery dialog option?
> 
> So for your usecase you would want a switch to make the full recovery window pop up instead of showing only the notification in the notifications window?

hello, could you make it with subfolder tree?

**Comment by diieg707** [comment]
![Image 2](https://user-images.githubusercontent.com/54504707/129349556-0ca87edb-a785-4e3f-8516-1f55a57abdc5.jpg)

kinda like this.
1. add a subfolder tree
2. if we select a subfolder all files within it will be set to be recovered
3. and add a checkmark to select ALL FILES to be recovered

**Comment by diieg707** [comment]
![129324542-bd4c65ba-962c-489b-8dfdfdfaa-a104dda21514](https://user-images.githubusercontent.com/54504707/129351891-32a2602b-f07b-41ae-bcf2-5154450dcef2.jpg)
my idea was to make the immediate recovery window wider (just like the full recovery in plus), with the list of destination folders on the botton

**Comment by isaak654** [comment]
Could you avoid to create more than two consecutive posts with attached screens in future, please?

**Comment by diieg707** [comment]
> 
> 
> Could you avoid to create more than two consecutive posts with attached screens in future, please?

sorry mate

**Comment by diieg707** [comment]
> 
> 
> How is this:
> ![grafik](https://user-images.githubusercontent.com/3890945/129324542-bd4c65ba-962c-489b-8faa-a104dda21514.png)
> 
> better than the current scheme?
> are you missing the ability to recover all the files at once?
> 
> so you want the behavioure of clicking the recovery -> open recovery dialog option?
> 
> So for your usecase you would want a switch to make the full recovery window pop up instead of showing only the notification in the notifications window?

answering your question yes, i would want the immediate recovery window back instead of only notification (because it sucks to recover lots of files at once), and more importantly the subtree folder

**Comment by diieg707** [comment]
hello, what about the list of destination folders?

![Image 1](https://user-images.githubusercontent.com/54504707/130458150-231d27ee-bbba-4a9e-a91d-cb69b9adaf7d.jpg)

**Comment by diieg707** [comment]
hello, the list of destination folders is missing in this prompt

![fixxx](https://user-images.githubusercontent.com/54504707/130458418-807b1c39-1239-4ea4-b867-32be666ef45b.jpg)

**Comment by diieg707** [comment]
> 
> 
> How is this:
> ![grafik](https://user-images.githubusercontent.com/3890945/129324542-bd4c65ba-962c-489b-8faa-a104dda21514.png)
> 
> better than the current scheme?
> are you missing the ability to recover all the files at once?
> 
> so you want the behavioure of clicking the recovery -> open recovery dialog option?
> 
> So for your usecase you would want a switch to make the full recovery window pop up instead of showing only the notification in the notifications window?

like this one

**Comment by diieg707** [comment]
> How is this:
> ![grafik](https://user-images.githubusercontent.com/3890945/129324542-bd4c65ba-962c-489b-8faa-a104dda21514.png)
> 
> better than the current scheme?
> are you missing the ability to recover all the files at once?
> 
> so you want the behavioure of clicking the recovery -> open recovery dialog option?
> 
> So for your usecase you would want a switch to make the full recovery window pop up instead of showing only the notification in the notifications window?

hello @DavidXanatos could you restore the immediate recovery to this window please? this one you posted with destination folder list

**Comment by DavidXanatos** [comment]
So you want teh recovery window to list fodlers used recently to recover to?

**Comment by diieg707** [comment]
> So you want teh recovery window to list fodlers used recently to recover to?

Yes,
And what about the #987 subfolder request?

**Comment by isaak654** [comment]
I don't think we can afford more users to downgrade to Classic UI because of this annoying Immediate recovery debate... things like this could affect bug reporting and future development:

> Thank you again for your reply. Yes, I can use "quick recovery" & recover multiple files at the same time. However, I can't do it with "immediate recovery". That feature is important enough to me that I think I'll go back to Sandboxie classic. 
Source: [Wilders](https://www.wilderssecurity.com/threads/immediate-recovery-doesnt-allow-multiple-files-at-once-in-sandboxie.440204/#post-3031857)

**Comment by diieg707** [comment]
> I don't think we can afford more users to downgrade to Classic UI because of this annoying Immediate recovery debate... things like this could affect bug reporting and future development:
> 
> > Thank you again for your reply. Yes, I can use "quick recovery" & recover multiple files at the same time. However, I can't do it with "immediate recovery". That feature is important enough to me that I think I'll go back to Sandboxie classic.
> > Source: [Wilders](https://www.wilderssecurity.com/threads/immediate-recovery-doesnt-allow-multiple-files-at-once-in-sandboxie.440204/#post-3031857)

The plus standard layout with notification is not good when it comes to recover lots of filescat once like extracting with winrar for example, so putting the classic layout would fix this problem

**Comment by DavidXanatos** [comment]
ok I'll make the recently added behavioure that uses the recovery window instead of the notifications one the default in the next build
about the list of where to recover to,
would it be ok if that would be a sub button in the recover to button instead of a full list?

**Comment by diieg707** [comment]
> ok I'll make the recently added behavioure that uses the recovery window instead of the notifications one the default in the next build
> about the list of where to recover to,
> would it be ok if that would be a sub button in the recover to button instead of a full list?

I don't know how you're planning to do it but as long as it has the list of destination folders it's ok

**Comment by DavidXanatos** [comment]
the idea is that its not a list in the window but the recover to button has a small arrow on the right and if you click on it it shows the list as a pop up menu you can than pick a destination and it starts recovering right aweay?
is that satisfying?

**Comment by diieg707** [comment]
> the idea is that its not a list in the window but the recover to button has a small arrow on the right and if you click on it it shows the list as a pop up menu you can than pick a destination and it starts recovering right aweay?
> is that satisfying?

Honestly I think that would be unpleasant for the user as it would require more effort to just show the destination list

**Comment by DavidXanatos** [comment]
![grafik](https://user-images.githubusercontent.com/3890945/132643469-1803955f-1e55-41b0-a2a8-96c5d84f741d.png)

isn't that great?
and as littel clicks as the other option

**Comment by diieg707** [comment]
> ![grafik](https://user-images.githubusercontent.com/3890945/132643469-1803955f-1e55-41b0-a2a8-96c5d84f741d.png)
> 
> isn't that great?
> and as littel clicks as the other option

I still thing it's not a good layout because if we mistakenly click a wrong destination folder it won't have the ability to correct

**Comment by diieg707** [comment]
Another suggestion to immediate recovery layout:
- add an option to make specific destination folder as default
that's useful when use immediate recovery window to restore a batch of files to same location

**Comment by diieg707** [comment]
> ![grafik](https://user-images.githubusercontent.com/3890945/132643469-1803955f-1e55-41b0-a2a8-96c5d84f741d.png)
> 
> isn't that great?
> and as littel clicks as the other option

The list of destination folders **up front** as in classic sbie is already an effective way to show destination folders

**Comment by diieg707** [comment]
Please reopen because I accidentaly closed it

**Comment by diieg707** [comment]
> > How is this:
> > ![grafik](https://user-images.githubusercontent.com/3890945/129324542-bd4c65ba-962c-489b-8faa-a104dda21514.png)
> > better than the current scheme?
> > are you missing the ability to recover all the files at once?
> > so you want the behavioure of clicking the recovery -> open recovery dialog option?
> > So for your usecase you would want a switch to make the full recovery window pop up instead of showing only the notification in the notifications window?
> 
> JUST LIKE THIS ONE

The only two major features immediate recovery window needs:
Subfolders tree, ( necessary to recover files in a folder at once, winrar)
And option to make specific destination folder as default

**Comment by DavidXanatos** [comment]
![grafik](https://user-images.githubusercontent.com/3890945/132739295-2e8caee1-6856-4dbf-96ae-bf0efcab5851.png)

**Comment by diieg707** [comment]
Great 👍,
So as I was saying your about your list idea  if we mistakenly click a wrong destination folder we won't have the chance to correct it

**Comment by DavidXanatos** [comment]
I could add a confirmation messagebox

**Comment by diieg707** [comment]
> I could add a confirmation messagebox

so i was also thinking to add a "remember" select box feature to keep a specific folder already selected when immediate recovery is called, that's why I preferred the here  https://github.com/sandboxie-plus/Sandboxie/issues/988#issuecomment-898268406

But if you don't like it it's ok

**Comment by DavidXanatos** [comment]
![grafik](https://user-images.githubusercontent.com/3890945/132744820-bb5cc7c5-f602-4992-abac-b6ac295cb460.png)
how about this?
Default combo box entry: same folder
then we have the list of otehr filders, and at the botom we have to brows for new folder

**Comment by diieg707** [comment]
> ![grafik](https://user-images.githubusercontent.com/3890945/132744820-bb5cc7c5-f602-4992-abac-b6ac295cb460.png)
> how about this?
> Default combo box entry: same folder
> then we have the list of otehr filders, and at the botom we have to brows for new folder
![images (12)](https://user-images.githubusercontent.com/54504707/132745678-fc6845e1-795b-41b5-9fd6-0d9855b4cb2e.jpeg)

Also add an option to remember specific destination folder, like in IDM download dialog

**Comment by diieg707** [comment]
> ![grafik](https://user-images.githubusercontent.com/3890945/132744820-bb5cc7c5-f602-4992-abac-b6ac295cb460.png)
> how about this?
> Default combo box entry: same folder
> then we have the list of otehr filders, and at the botom we have to brows for new folder

Also, considering that most programs have their "APPLY" button on the botton you could place the "Recover to"  button in the place of that  "delete all" button

**Comment by DavidXanatos** [comment]
what is this remember option ment to do? not remember each time somethign sustom is selected but only when the check is checked?

**Comment by diieg707** [comment]
what that "delete all" button do?

**Comment by diieg707** [comment]
> what is this remember option ment to do? not remember each time somethign sustom is selected but only when the check is checked?

remember option is to (remember specific destination folder as selected), useful for people constantly recovering files to the same location

**Comment by DavidXanatos** [comment]
aaaa i see

**Comment by diieg707** [comment]
> aaaa i see

👍

**Comment by DavidXanatos** [comment]
![grafik](https://user-images.githubusercontent.com/3890945/132752681-7e43a7a0-40df-40b5-886b-fc7200fe84b9.png)
happy?

**Comment by diieg707** [comment]
> ![grafik](https://user-images.githubusercontent.com/3890945/132752681-7e43a7a0-40df-40b5-886b-fc7200fe84b9.png)
> happy?

it seems you removed the option to clean destination folder history

**Comment by diieg707** [comment]
You could put recover button on left side of close button
and clean list on the place of the current recover button

**Comment by DavidXanatos** [comment]
clean list is already the last option in the list 
also in need the palce left of teh close button for the here not shown delete button when showing this dialog in place of the delete confirmation dialog

**Comment by diieg707** [comment]
ok then, that's more like it 💜👍

**Comment by diieg707** [comment]
I'm 99% happy, not 100 coz I preferred the list instead of a dropdown

**Comment by diieg707** [comment]
> ![grafik](https://user-images.githubusercontent.com/3890945/132752681-7e43a7a0-40df-40b5-886b-fc7200fe84b9.png)
> happy?

I'm happy now, great layout 💜👍

**Comment by diieg707** [comment]
dropdown vs list.

in a scenario that we need to recover to lots of different locations the dropdown is not as efficient as the list layout

the list layout would save us 1 click per additional location we need to select.
so after some thought concluded that dropdown list to recover to many different location takes 1 additional click compared to list layout

**Comment by diieg707** [comment]
> ![grafik](https://user-images.githubusercontent.com/3890945/132752681-7e43a7a0-40df-40b5-886b-fc7200fe84b9.png)
> happy?

So in conclusion I am 95% happy about the layout, not 100 because the dropdown layout takes 1 additional click (to open) in cases to recover to many different location

**Comment by diieg707** [comment]
After knowing this issue are you still keeping the dropdown or will change it to list?

dropdown: open and select (2 clicks) 
list: (1 click)

**Comment by diieg707** [comment]
Update:
don't forget the **"don't prompt again until all sandboxed programs stop"** checkbox in the immediate recovery layout

**Comment by diieg707** [comment]
Update:
don't forget the **"don't prompt again until all sandboxed programs stop"** checkbox
![images (2)](https://user-images.githubusercontent.com/54504707/132935878-10e5b621-67f0-4a03-ae23-d50e91616dd5.png)

**Comment by DavidXanatos** [comment]
I will add the don't prompt button.

About the drop down, I really don't like the list as it takes away to much from the file tree.
I will think about if i can make it look good to have a small arrow button that expands the drop down to a full list, 
and that would remember the option for the next run.

**Comment by DavidXanatos** [comment]
Or you know what if you want that list sooo soo much, than become a patreon on one of the tears that give you small custom features ;)

**Comment by diieg707** [comment]
> I will add the don't prompt button.
> 
> About the drop down, I really don't like the list as it takes away to much from the file tree.
> ~I will think about if i can make it look good to have a small arrow button that expands the drop down to a full list, and that would remember the option for the next run.~

I understand, I was just pointing out all the possibilities in order to make an effective layout

**Comment by diieg707** [comment]
can you make the immediate recovery auto close once all files in its windows are recovered?

**Comment by DavidXanatos** [comment]
I don't think this is ideal, the user may after having recovered the files want to click the show all files check box to inspect other files.

**Comment by diieg707** [comment]
> 
> 
> I don't think this is ideal, the user may after having recovered the files want to click the show all files check box to inspect other files.

i see, or maybe add a checkbox to close window after all files are recovered?

**Comment by diieg707** [comment]
> 
> 
> I don't think this is ideal, the user may after having recovered the files want to click the show all files check box to inspect other files.

I have an workaround idea to cover both cases:
- add the checkbox to auto close when all are recovered
- IF the show all files box is marked then the auto close checkbox is automatically unchecked

also: 
if the immediate recovery is called leave autoclose checked by default
if recovery window is called from sbie panel/when delete files autoclose checkbox is unchecked by default
